### PR TITLE
Added HelloWorld, NEP5 example, Domain, and Lock SC using latest java-framework 2.3.0

### PR DESCRIPTION
**Pull request by fabwa**

### CODE REVIEW

**Comment by erikzhang** [comment]
Do you mean NEP-5?

**Comment by fabwa** [comment]
Made amendments to support required nep5 calls. Renamed

File: Domain/Domain.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+        if (operation == "query") return Query((String) args[0]);
+        if (operation == "register") return Register((String)args[0], (byte[])args[1]);
+        if (operation == "transfer") return Transfer((String) args[0], (byte[])args[1]);
+        if (operation == "delete") return Delete((String) args[0]);
+        return false;
+
+    }
+
+    private static byte[] Query(String domain){
```
Comment (privettoli) [inline Domain/Domain.java]: names of methods should be in lowerCamelCase according to Java Code Convention

File: ERC20/NEP5TOK.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import org.neo.smartcontract.framework.SmartContract;
+import org.neo.smartcontract.framework.services.neo.Runtime;
+import org.neo.smartcontract.framework.services.neo.Storage;
+
+public class NEP5TOK extends SmartContract {
+
+    public static Object Main(byte[] originator, String Event, byte[] args0, byte[] args1, byte[] args2) {
+        
+    	BigInteger supply = BigInteger.valueOf(100000);
+        String name = "NEP5TOK";
```
Comment (privettoli) [inline ERC20/NEP5TOK.java]: move these constants to private static final String CONSTANT with meaningful name

File: ERC20/NEP5TOK.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        String sym = "N5T";
+        byte decimals = 0;
+        
+        if (!Runtime.checkWitness(originator)) return false;
+
+        if (Event == "totalSupply") return supply;
```
Comment (privettoli) [inline ERC20/NEP5TOK.java]: move those ifs into
enum EventName {
NAME("name"),
SYMBOL("symbol"),
…
}
and
Map<EventName, EventHandler>
where EventHandler is an interface that contain result of function (or nothing)
and then move all methods into EventHandler implementations

File: ERC20/NEP5TOK.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        if (Event == "name") return name;
+        if (Event == "symbol") return sym;
+        if (Event == "decimals") return decimals;
+        if (Event == "deploy") return Deploy(originator, supply);
+        if (Event == "balanceOf") return Storage.get(Storage.currentContext(), args0);
+        if (Event == "transfer") return Transfer(originator, args0, args1);
+        if (Event == "transferFrom") return TransferFrom(originator, args0, args1, args2);
+        if (Event == "approve") return Approve(originator, args0, args1);
+        if (Event == "allowance") return Allowance(args0, args1);
+        return false;
```
Comment (privettoli) [inline ERC20/NEP5TOK.java]: move magic boolean false into private static final boolean MEANINGFUL_NAME

File: ERC20/NEP5TOK.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        }
+        return false;
+    }
+
+    private static Boolean TransferFrom(byte[] originator, byte[] from, byte[] to, byte[] amount) {
+        BigInteger allVallong = new BigInteger(Storage.get(Storage.currentContext(), Helper.concat(from, to)));
+        BigInteger fromVallong = new BigInteger(Storage.get(Storage.currentContext(), from));
+        BigInteger toVallong = new BigInteger(Storage.get(Storage.currentContext(), to));
+
+        if (fromVallong.compareTo(BigInteger.ZERO) != -1 && new BigInteger(amount).compareTo(BigInteger.ZERO) != -1 &&
```
Comment (privettoli) [inline ERC20/NEP5TOK.java]: do not use compareTo, use BigInteger.ZERO.equals(variable)

File: Lock/lock.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import org.neo.smartcontract.framework.services.neo.Blockchain;
+import org.neo.smartcontract.framework.services.neo.Header;
+
+
+public class lock extends SmartContract{
+	
+    public static Boolean Main(int timestamp, byte[] pubkey, byte[] signature)
+    {
+        Header header = Blockchain.getHeader(Blockchain.height());
+        if (timestamp > header.timestamp()) return false;
```
Comment (privettoli) [inline Lock/lock.java]: according to Java Code Convention it's cleaner to have parentheses on ALL conditions

File: Domain/Domain.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import org.neo.smartcontract.framework.SmartContract;
+import org.neo.smartcontract.framework.services.neo.Storage;
+
+public class Domain extends SmartContract {
+
+    public static Object Main(String operation, Object... args){
+
+        if (operation == "query") return Query((String) args[0]);
```
Comment (privettoli) [inline Domain/Domain.java]: do same refactoring as for NEP5TOK

File: Domain/Domain.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        if (operation == "register") return Register((String)args[0], (byte[])args[1]);
+        if (operation == "transfer") return Transfer((String) args[0], (byte[])args[1]);
+        if (operation == "delete") return Delete((String) args[0]);
+        return false;
```
Comment (privettoli) [inline Domain/Domain.java]: do same refactoring as for NEP5TOK

File: ERC20/NEP5TOK.java [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        if (Event == "transfer") return Transfer(originator, args0, args1);
+        if (Event == "transferFrom") return TransferFrom(originator, args0, args1, args2);
+        if (Event == "approve") return Approve(originator, args0, args1);
+        if (Event == "allowance") return Allowance(args0, args1);
+        return false;
+    }
+    
+    private static Boolean Deploy(byte[] originator, BigInteger supply) {
+        byte[] adminKey = new byte[] {};
+        if (originator != adminKey) return false;
```
Comment (privettoli) [inline ERC20/NEP5TOK.java]: impossible condition, create static ByteUtils.isEmpty with check of null and array.isEmpty

File: ERC20/NEP5TOK.java [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+        Storage.put(Storage.currentContext(), originator, supply.toByteArray());
+        return true;
+    }
+
+    private static Boolean Transfer(byte[] originator, byte[] to, byte[] amount) {
+        BigInteger originatorValue = new BigInteger(Storage.get(Storage.currentContext(), originator));
+        BigInteger targetValue = new BigInteger(Storage.get(Storage.currentContext(), to));
+
+        BigInteger nOriginatorValue = originatorValue.subtract(new BigInteger(amount));
```
Comment (privettoli) [inline ERC20/NEP5TOK.java]: what does "n" means? name it meaningfully please

File: Domain/Domain.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
+import org.neo.smartcontract.framework.SmartContract;
```
Comment (privettoli) [inline Domain/Domain.java]: format all files by Java Code Convention (or use IntelliJ Idea to format it)

File: Lock/lock.java [hunk 12]
```diff
@@ -0,0 +1,14 @@
+import org.neo.smartcontract.framework.SmartContract;
+import org.neo.smartcontract.framework.services.neo.Blockchain;
+import org.neo.smartcontract.framework.services.neo.Header;
+
+
+public class lock extends SmartContract{
```
Comment (privettoli) [inline Lock/lock.java]: the classname should be in UpperCamelCase (Lock)

File: Domain/Domain.java (hunk 11 above)
Comment (fabwa) [reply Domain/Domain.java]: thanks a lot for your suggestions! While they all make sense in the java world, some are not yet supported by the neo VM (events, maps, enum). Will amend the style recommendations.

File: Lock/lock.java (hunk 12 above)
Comment (notatestuser) [reply Lock/lock.java]: @privettoli you are like a human Java linter!

**Comment by mpetrunic** [comment]
@fabwa Did you actually test(deploy and invoke) this code? As far as i know you can only send hex strings as params so you can't actually  send these operation names

**Comment by fabwa** [comment]
@mpetrunic sure i have. Check for example 4bc67dbd6b374d179a9b1869add111e6a3b12ea3 on testnet.
https://i.imgur.com/k0CKTEb.jpg
Using NEO-GUI you can pass different parameters

**Pull request closed without merging**